x-pack/plugins/features/common/feature.ts

```diff
@@ -98,7 +98,17 @@ export interface Feature<
    * ```
    * @see FeatureKibanaPrivileges
    */
-  privileges: TPrivileges;
+  privileges:
+    | {
+        all: FeatureKibanaPrivileges;
+        read: FeatureKibanaPrivileges;
+      }
+    | 'none';
```

Comment:
```suggestion
    | null;
```

Using the string `'none'` seems rather awkward here. Especially since all of the features automatically get the "none" privileges in the UI. Using `null` here instead requires that the consumers specify `null` instead of just "forget" to specify privileges, and it also lets us use TS's nullish coalescing.

Reply:
Good suggestion, I like this change.